### PR TITLE
Gh enterprise 1098

### PR DESCRIPTION
**Pull request by danantonioreyes**
**Changes**

- Added model EmailInvite
- Made changes to teams.forms.InviteForm to process the email address field
- Added an email field to the invite member page

**Testing**
- go to "Members" page of a team then go to "Invite Member"
- you can use [Mailinator](https://www.mailinator.com/) for a dummy inbox (for example, unisubstestuser@mailinator.com)

**Spill-overs**

1. Update `templates/new-teams/invite.html` to use future ui -- there is already an Invite member template that uses future UI. It just needs more work to make the username autocomplete field work, and to render the invite message field properly
2. Update `templates/new-teams/invite.html` to enable multiple username-role and email-role combinations (i.e. the team owner/admin can invite multiple users and email addresses each with its own role)
3. The signup screen should enable users to sign up with their Google, FB, and other OAuth accounts

### CODE REVIEW

**Comment by danantonioreyes** [comment]
This should be QA-testable code now.

File: apps/teams/new_views.py [hunk 1]
```diff
@@ -576,6 +587,41 @@ def invite(request, team):
...
     })
 
+def email_invite(request, signed_pk):
+    try:
+        pk = EmailInvite.signer.unsign(signed_pk)
+        email_invite = EmailInvite.objects.get(pk=pk)
+
+        form = CustomUserCreationForm(request.POST or None, label_suffix="")
+
+        if(request.POST):
```
Comment (bendk) [inline apps/teams/new_views.py]: I think it's better to use request.method == 'POST'.  It's possible to send data with a GET request, and django will put that in the POST dict which will make this test pass.

This isn't really a big deal, but I think it's a bit better to check the method than the request data.

**Comment by bendk** [comment]
Looks great overall, I just had one small suggestion.  Thanks Dan!

**Comment by danantonioreyes** [comment]
Thanks for the review Ben! Implemented the suggestions in the most recent commit

File: apps/teams/forms.py [hunk 2]
```diff
@@ -1006,9 +1006,13 @@ def save(self):
...
 class InviteForm(forms.Form):
-    username = UserAutocompleteField(error_messages={
-        'invalid': _(u'User has a pending invite or is already a member of this team'),
-    })
+    username = UserAutocompleteField(required=False, error_messages={
+        'invalid': _(u'User has a pending invite or is already a member of this team')
+        },
+        help_text="Amara username of the user you want to invite")
+    email = forms.EmailField(required=False, max_length=254,
+                            widget=forms.TextInput(attrs={'width':'100%'}),
```
Comment (bendk) [inline apps/teams/forms.py]: I don't think you should have to add the `width: 100%` thing here.  Was there an issue with the field styling?

Comment (danantonioreyes) [reply apps/teams/forms.py]: It seems to be that when I initially tested without specifying the widget, it was defaulting to a forms.EmailInput which was not rendering the textbox through the entirety of the width of the form.

Pre-emptively added width 100% to make it render correctly, but have confirmed that just specifying a forms.TextInput widget without the width is good enough

**Pull request merged**